packages/xsnap/src/xsnap.js

```diff
@@ -293,16 +330,31 @@ export function xsnap(options) {
   }
 
   /**
-   * @param {string} file
-   * @returns {Promise<void>}
+   * @param {string} [hint]
+   * @returns {AsyncGenerator<Uint8Array, void, undefined>}
    */
-  async function writeSnapshot(file) {
-    const result = baton.then(async () => {
-      await messagesToXsnap.next(encoder.encode(`w${file}`));
-      await runToIdle();
+  async function* makeSnapshot(hint = 'unknown') {
```

Comment:
maybe `makeSnapshotReader` or `makeSnapshotReadStream`? although it doesn't feel as urgent for this case, somehow